### ISSUE TITLE
Code content does not match the paper!

### ISSUE THREAD

**Issue by surpoloyang** [Huang2002200/PlantNet-and-PSegNet#4]
Firstly, thank you for the your's open-source spirit!
During the review of the Plantnet_pytorch version code, I found that the definition of the DHL loss function in the code for case1 and case3 is exactly opposite to that in the paper, as follows:
![image](https://github.com/user-attachments/assets/b87143e2-60fc-4487-a216-3219add2cd5b)

The case 1 in the paper refers to situations with different semantic classes and different instance classes, whereas the corresponding code deals with the same instances (this is the case 3 in the paper), which leads to different calculation results for d(i,j) and further results in different calculations for the DHL loss function, causing me great confusion, not knowing whether to follow the description in the paper or the presentation in the code 😕
![image](https://github.com/user-attachments/assets/c37dcd88-0ef3-4ee5-8377-b757973031ae)

**Comment by Huang2002200** [comment]
Dear surpoloyang,

Thank you for pointing out this misalignment. The code of PlantNet and PSegNet has been passed down and modified through several graduate students and researchers. Therefore, the current code version may not be strictly the same with the original paper. In the recent two years, we have applied/tested the two networks on various kinds of 3D datasets including crops, CAD models, street views. And during the application, we might have altered a few realizations and parameters to find the optimal solution, which results in the misalignment.

In order to objectively respond to your concern, we compared the code version in this repo and the version strictly realized according to the original paper (at least we think it is more strict than the last one) on the same crop dataset. The comparison was done for training of 200 epochs, and the quantitative results for segmentation on the same testing set are quite similar (the result of comparison can be found in the following figure). To sum up, we think both versions are fine.

We will modify the code in this repo to the paper version later.

Our best.


![Image](https://github.com/user-attachments/assets/7913753c-7834-49fe-8b3e-5403b97bb7e3)

**Comment by surpoloyang** [comment]
Thank you for your rigorous research style！
